### PR TITLE
Add pretrained Detrectron2 model 

### PR DESCRIPTION
**Pull request by haossr**
**Problem**

- Starter misses basic detection models.

**Solution**

- Added the pretrained detection models from Detrectron2.
- Updated model unit test.
- 

**Tests**

Covered by CircleCI.

### CODE REVIEW

File: models/detection.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+MODEL_CONFIG = {"faster_rcnn": "COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml",
+                "retinanet": "COCO-Detection/retinanet_R_50_FPN_3x.yaml",
+                "rpn": "COCO-Detection/rpn_R_50_FPN_1x.yaml",
+                "fast_rcnn": "COCO-Detection/fast_rcnn_R_50_FPN_1x.yaml",
+               }
+
+class DetectionModel(nn.Module):
+    pass
```
Comment (jirvin16) [inline models/detection.py:17]: Will this be used for other models?

File: test/test_model.py [hunk 2]
```diff
@@ -1,11 +1,17 @@
 import unittest
 
-from models import CadeneModel, TorchVisionModel
+from models import (CadeneModel, 
+                    TorchVisionModel, 
+                    EfficientNetModel,
```
Comment (jirvin16) [inline test/test_model.py:5]: I assume this is going to be implemented in another PR?

**Review by jirvin16 — APPROVED** [review]
LGTM!

Seems one tricky thing here is to make sure bounding boxes are formatted correctly as input to each model and parsed correctly when output by each model. Does it make sense to modify the forward pass to do this? Or have helper functions before loss calculation and after forward pass to normalize this across models?

File: models/detection.py (hunk 1 above)
Comment (haossr) [reply models/detection.py:17]: Yep -- want to have a parent class for Detrectron2, YOLO, and EfficientNet later.

File: test/test_model.py (hunk 2 above)
Comment (haossr) [reply test/test_model.py:5]: Oh, this is to update the unit test to include Detrectron2 models (also add efficientnet which is previously missed)

**Comment by haossr** [comment]
Yeah I think the format is going to be tricky. I am leaning towards to setting detectron2 [model input format](https://detectron2.readthedocs.io/en/latest/tutorials/models.html#model-input-format) as default -- They have a nice [dataset mapper class](https://detectron2.readthedocs.io/en/latest/modules/data.html#detectron2.data.DatasetMapper) that will make it much easy to load and augment the images/annotations.  
I am planning to align Yolo and EfficientNet's input format to that (aka parse the format inside the model wrapper in `model/detection.py` here) later. Let me know if that does not make too much sense to you.

File: test/test_model.py (hunk 2 above)
Comment (jirvin16) [reply test/test_model.py:5]: Ah this is `EfficientNet`, thought it said `EfficientDet`. Nice catch!

**Comment by jirvin16** [comment]
> Yeah I think the format is going to be tricky. I am leaning towards to setting detectron2 [model input format](https://detectron2.readthedocs.io/en/latest/tutorials/models.html#model-input-format) as default -- They have a nice [dataset mapper class](https://detectron2.readthedocs.io/en/latest/modules/data.html#detectron2.data.DatasetMapper) that will make it much easy to load and augment the images/annotations.
> I am planning to align Yolo and EfficientNet's input format to that (aka parse the format inside the model wrapper in `model/detection.py` here) later. Let me know if that does not make too much sense to you.

That makes sense to me!
